### PR TITLE
Fix typos in fleet/docs/architecture.md

### PR DESCRIPTION
**Pull request by viccuad**

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -21,25 +21,25 @@ connections to downstream clusters. This means managed clusters can run in priva
...
-The Fleet manager dynamically creates service account, manages their RBAC and then gives the
+The Fleet manager dynamically creates service accounts, manages their RBAC and then gives the
 tokens to the downstream clusters. Clusters are registered by optionally expiring cluster registration tokens.
 The cluster registration token is used only during the registration process to generate a credential specific
 to that cluster. After the cluster credential is established the cluster "forgets" the cluster registration
  token.
 
 The service accounts given to the clusters only have privileges to list `BundleDeployment` in the namespace created
-specifically for that cluster.  It can also update the `status` subresource of `BundleDeployment` and the `status`
+specifically for that cluster. They can also update the `status` subresource of `BundleDeployment` and the `status`
```
Comment (nickgerace) [inline docs/architecture.md:39]: This is referring to multiple service accounts, and `It` is fine.

**Review by nickgerace — CHANGES_REQUESTED** [review]

**Comment by nickgerace** [comment]
Thank you for your PR!

File: docs/architecture.md [hunk 2]
```diff
@@ -21,25 +21,25 @@ connections to downstream clusters. This means managed clusters can run in priva
 NATs. The only requirement is the cluster agent needs to be able to communicate with the
 Kubernetes API of the cluster running the Fleet manager. The one exception to this is if you use
 the [manager initiated](./manager-initiated.md) cluster registration flow.  This is not required, but
-an optional patter.
```
Comment (nickgerace) [inline docs/architecture.md:24]: This is fixed by: https://github.com/rancher/fleet/pull/229
We may want to wait for that PR to be merged, and then rebase.

File: docs/architecture.md (hunk 1 above)
Comment (viccuad) [reply docs/architecture.md]: Dropping it then :).

File: docs/architecture.md (hunk 2 above)
Comment (viccuad) [reply docs/architecture.md]: Done in https://github.com/rancher/fleet/pull/227/commits/e4c539903c7c4c9f3c053705ac8b4eee63e723a8.

**Comment by viccuad** [comment]
Rebased on top of current master and tackled review.

Please feel free to close the PR and contribute the typo corrections on your own if that makes it faster :).

**Review by nickgerace — APPROVED** [review]

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**